x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/edit_policy.test.ts

```diff
@@ -672,66 +672,155 @@ describe('<EditPolicy />', () => {
+          });
+          test('hiding default, recommended option', async () => {
+            const { actions, find } = testBed;
+            await actions.warm.enable(true);
+            actions.warm.showDataAllocationOptions();
+
+            expect(find('defaultDataAllocationOption').exists()).toBeFalsy();
+            expect(find('customDataAllocationOption').exists()).toBeTruthy();
+            expect(find('noneDataAllocationOption').exists()).toBeTruthy();
           });
-
-          const { component } = testBed;
-          component.update();
         });
-        test('defaults searchable snapshot to true on cloud', async () => {
-          const { find, actions } = testBed;
-          await actions.cold.enable(true);
-          expect(
-            find('searchableSnapshotField-cold.searchableSnapshotToggle').props()['aria-checked']
-          ).toBe(true);
+
+        describe('v8 and above', () => {
+          beforeEach(async () => {
+            httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('my_policy')]);
+            httpRequestsMockHelpers.setListNodes({
+              nodesByAttributes: { test: ['123'] },
+              // On cloud, even if there are data_* roles set, the default, recommended allocation option should not
+              // be available.
+              nodesByRoles: { data_hot: ['123'] },
+              isUsingDeprecatedDataRoleConfig: true,
+            });
+            httpRequestsMockHelpers.setListSnapshotRepos({ repositories: ['found-snapshots'] });
+
+            await act(async () => {
+              testBed = await setup({
+                appServicesContext: {
+                  cloud: {
+                    isCloudEnabled: true,
+                  },
+                  license: licensingMock.createLicense({ license: { type: 'basic' } }),
+                  stackVersion: '8.0.0',
+                },
+              });
+            });
+
+            const { component } = testBed;
+            component.update();
+          });
+          test('hiding default, recommended option', async () => {
+            const { actions, find } = testBed;
+            await actions.warm.enable(true);
+            actions.warm.showDataAllocationOptions();
+
+            // Should show all options!!
+            expect(find('defaultDataAllocationOption').exists()).toBeTruthy();
+            expect(find('customDataAllocationOption').exists()).toBeTruthy();
+            expect(find('noneDataAllocationOption').exists()).toBeTruthy();
+          });
         });
       });
-      describe('existing policy', () => {
-        beforeEach(async () => {
-          httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('my_policy')]);
-          httpRequestsMockHelpers.setListNodes({
-            isUsingDeprecatedDataRoleConfig: false,
-            nodesByAttributes: { test: ['123'] },
-            nodesByRoles: { data: ['123'] },
+      describe('using data role config', () => {
```

Comment:
I found it difficult to discern the difference between the tests on line 798 and 810 and the tests on 703 and 741. Could we change this description to "using node roles"? Is that an accurate description? If so, I would find it easier to contrast that against the description on line 676 of "using legacy data role config".